### PR TITLE
Skip tests that rely on DB if MySQL not being used

### LINKED ISSUE #37: Module (currently) only works with MySQL, but does not guard it's tests properly
When using another adapter to speed up tests (such as `silverstripe/sqlite3`) this results in a set of false failures. The tests should be marked as skipped instead.

### PR DESCRIPTION
**Pull request by dhensby**
fixes #37

### CODE REVIEW

**Comment by dhensby** [comment]
Interesting, we have test failures against the hybrid store that are not DB driver related - do we have a bug?

**Comment by NightJar** [comment]
assertEquals is doing strict type checking in PHP 7.2 it would seem.
But then that begs the question of if it's `===` why does it work in other versions?

**Comment by dhensby** [comment]
I don't think it's 7.2 vs 7.1 tbh; `assertEquals()` has always been strict and `assertSame` is loose.

We unserialize the value so it should come out the same... we aren't try to assert `[falsey] == [falsey] in this test, it's that an `[array with data] == [array with data]`

**Comment by NightJar** [comment]
Ah hah.

HybridSession is made to use CookieStore, then fall back to DatabaseStore when the former doesn't cut it.

Our fatal mistake was to assume that the CookieStore tests were also just the AbstractTest with the getStore method implemented. However there are actually four tests for CookieStore, and it overrides the `testStoreLargeData` test, because it's *supposed* to fail!
CookieStore does not (and cannot) store large information, so the whole idea of this test for HybridSession is to fall back to storing large things in the database.

Removing the database layer breaks that assumption and causes the test to fail. So again if MySQL isn't present it needs a skip, becasuse no data is _never_ going to suddenly be _some_ data (hur hurr heartblead)

File: .travis.yml [hunk 1]
```diff
@@ -10,13 +10,16 @@ matrix:
...
+    - php: 7.1
+      env: DB=PGSQL PHPUNIT_TEST=1
 
 before_script:
   - phpenv rehash
   - phpenv config-rm xdebug.ini
   - echo 'memory_limit = 2G' >> ~/.phpenv/versions/$(phpenv version-name)/etc/conf.d/travis.ini
   - composer validate
   - composer require silverstripe/recipe-core 1.0.x-dev --no-update
+  - if [[ $DB == PGSQL ]]; then composer require silverstripe/postgresql:2.0.x-dev --no-update; fi
```
Comment (robbieaverill) [inline .travis.yml]: This will need to be 2.1.x-dev now for transaction support

**Comment by robbieaverill** [comment]
These tests aren't very nice anyway. The HybridSession class deserves some actual unit tests I reckon

**Comment by robbieaverill** [comment]
@dhensby I've refactored the HybridSession unit tests - see what you think, you're more than welcome to reset and wipe out my commits =)

**Comment by robbieaverill** [comment]
@dhensby I've rebased this branch FYI

**Comment by dhensby** [comment]
well, we are green - I won't ask too many questions :)

**Pull request merged**